src/main/java/dk/cngroup/lentils/controller/HintController.java

```diff
@@ -49,40 +56,58 @@ public String deleteHint(@PathVariable("hintId") final Long hintId) {
         Hint hint = hintService.getHint(hintId);
         Long cypherId = hint.getCypherId();
         hintService.deleteById(hintId);
-        return REDIRECT_HINT_LIST + "?cypherId=" + cypherId;
+        return REDIRECT_HINT_LIST + CYPHERID_PARAMETER + cypherId;
     }
 
     @GetMapping(value = "/update/{hintId}")
     public String addForm(@PathVariable("hintId") final Long hintId, final Model model) {
-        Hint hint = hintService.getHint(hintId);
+        HintFormDTO hintFormDto = mapper.map(hintService.getHint(hintId), HintFormDTO.class);
         model.addAttribute(TEMPLATE_ATTR_HEADING, "Upravit hint");
-        model.addAttribute(TEMPLATE_ATTR_HINT, hint);
+        model.addAttribute(TEMPLATE_ATTR_HINT, hintFormDto);
         return VIEW_HINT;
     }
 
+    @PostMapping(value = "/update/{hintId}")
+    public String save(@PathVariable("hintId") final Long hintId,
+                       @Valid @ModelAttribute("hint") final HintFormDTO hintFormDto,
+                       final BindingResult bindingResult,
+                       final Model model) {
+        if (bindingResult.hasErrors()) {
+            model.addAttribute(TEMPLATE_ATTR_HEADING, getHintHeading(hintFormDto));
+            model.addAttribute(TEMPLATE_ATTR_HINT, hintFormDto);
+            return VIEW_HINT;
+        }
+
+        Hint hint = mapper.map(hintFormDto, Hint.class);
+        hint.setHintId(hintId);
```

Comment:
Co když pošlu update na neexistující ID? Vytvoří to entitu s tímhle ID a ještě nenavázanou na šifru?

Udělal bych to tak, že udělých get(id),. zkontroluju jestli je v databázi a pak teprv namapuju a uložím.